### PR TITLE
Get bunx tests to pass on Windows

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

- Fixes "unable to find executable" error caused by checking the last write time of `<bunx-dir>/node_modules/uglifyjs/package.json` rather than `<bunx-dir>/package.json`
- _should_ fix `SharingViolation` error
- Should fix flakiness from deleting tmp dirs, I think

### How did you verify your code works?

Tests

### CODE REVIEW

File: src/windows.zig [hunk 1]
```diff
@@ -3022,7 +3022,12 @@ pub fn translateNTStatusToErrno(err: win32.NTSTATUS) bun.C.E {
         .OBJECT_NAME_NOT_FOUND => .NOENT,
         .NOT_A_DIRECTORY => .NOTDIR,
         .RETRY => .AGAIN,
+        .DIRECTORY_NOT_EMPTY => .EXIST,
         .FILE_TOO_LARGE => .@"2BIG",
+        .SHARING_VIOLATION => if (comptime Environment.isDebug) brk: {
+            bun.Output.debugWarn("Received SHARING_VIOLATION, indicates file handle should've been opened with FILE_SHARE_DELETE", .{});
```
Comment (paperclover) [inline src/windows.zig:3046]: glad to see this was found.

File: src/bun.js/node/node_fs.zig [hunk 2]
```diff
@@ -4812,7 +4812,7 @@ pub const NodeFS = struct {
         const fd = switch (switch (Environment.os) {
             else => Syscall.openat(atfd, basename, flags, 0),
             // windows bun.sys.open does not pass iterable=true,
-            .windows => bun.sys.openDirAtWindowsA(atfd, basename, true, false),
+            .windows => bun.sys.openDirAtWindowsA(atfd, basename, .{ .no_follow = true }),
```
Comment (dylan-conway) [inline src/bun.js/node/node_fs.zig]: is this supposed to be `.iterable = true`

File: src/bun.zig [hunk 3]
```diff
@@ -2347,14 +2367,140 @@ pub inline fn OSPathLiteral(comptime literal: anytype) *const [literal.len:0]OSP
...
+        }
+
+        return self.makeOpenPath(sub_path, opts);
+    }
+
     /// copy/paste of `std.fs.Dir.makePath` and related functions and modified to support u16 slices.
     /// inside `MakePath` scope to make deleting later easier.
     /// TODO(dylan-conway) delete `MakePath`
     pub fn makePath(comptime T: type, self: std.fs.Dir, sub_path: []const T) !void {
+        if (Environment.isWindows) {
```
Comment (jdalton) [inline src/bun.zig:2517]: comptime'd already?

File: src/install/install.zig [hunk 4]
```diff
@@ -1275,62 +1271,137 @@ pub const PackageInstall = struct {
...
+
         const FileCopier = struct {
             pub fn copy(
                 destination_dir_: std.fs.Dir,
                 walker: *Walker,
                 progress_: *Progress,
+                to_copy_into1: if (Environment.isWindows) []u16 else void,
+                head1: if (Environment.isWindows) []u16 else void,
+                to_copy_into2: if (Environment.isWindows) []u16 else void,
+                head2: if (Environment.isWindows) []u16 else void,
```
Comment (jdalton) [inline src/install/install.zig:1390]: comptime?

File: src/bun.zig (hunk 3 above)
Comment (nektro) [reply src/bun.zig:2517]: yep

**Comment by jdalton** [comment]
There's is a lot changed here. Hard to reason about all of it (means I defer to unit tests and CI happiness), but would also welcome any context/note comments.

File: src/install/repository.zig [hunk 5]
```diff
@@ -308,14 +297,9 @@ pub const Repository = extern struct {
...
-            var dir = try bun.openDir(cache_dir, folder_name);
+            const folder = Path.joinAbsString(PackageManager.instance.cache_directory_path, &.{folder_name}, .auto);
 
-            cwd = if (comptime Environment.isWindows)
-                Path.joinAbsString(PackageManager.instance.cache_directory_path, &.{folder_name}, .windows)
-            else
-                dir;
-
-            _ = exec(allocator, env, cwd, &[_]string{ "git", "checkout", "--quiet", resolved }) catch |err| {
+            _ = exec(allocator, env, "", &[_]string{ "git", "-C", folder, "checkout", "--quiet", resolved }) catch |err| {
```
Comment (Jarred-Sumner) [inline src/install/repository.zig]: clever

**Pull request merged**